lib/publiccloud/azure.pm

```diff
@@ -249,24 +254,29 @@ sub generate_tags {
     my $definition = $self->generate_azure_image_definition();
 
 Generated the Azure Image name from the job settings. If present, it takes the PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION setting.
-If not present it generated the image definition name based on distri, version, flavor and SKU.
+If not present, it generates the image definition name based on distri, version, flavor and SKU.
 
+    input $type: 'undefined/null/empty' to provide full image name; ='base' to get the base-name excluding 'sku'.
 Note: Image definitions needs to be distinct names and can only serve one architecture!
 
 Example: 'SLE-MICRO-5.4-BYOS-AZURE-X86_64-GEN2'
+
+    B<return> a string with the image full name distri-version-flavor-arch-sku, if $type undefined; 
+              otherwise the base-name without '-sku'.
 =cut
 
 sub generate_azure_image_definition {
-    my ($self) = @_;
+    my ($self, $type) = @_;
     return get_var('PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION') if (get_var('PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION'));
 
-    my $distri = get_required_var('DISTRI');
+    my $distri = (is_sle_micro('>=6.0')) ? 'sl-micro' : get_required_var('DISTRI');
     my $version = get_required_var('VERSION');
     my $flavor = get_required_var('FLAVOR');
-    my $arch = (get_var('PUBLIC_CLOUD_ARCH', get_required_var('ARCH'))) eq 'x86_64' ? 'x64' : 'Arm64';
+    my $arch = az_arch();
     my $sku = get_var('PUBLIC_CLOUD_AZURE_SKU', $default_sku);
-    my $image = uc("$distri-$version-$flavor-$arch-$sku");
-    return $image;
+    my $image = "$distri-$version-$flavor-$arch";
+    return uc($image) if (defined($type) and $type == 'base');
```

Comment:
if you would accept suggestion above this line could look like 
```suggestion
    return  ($basename)  ? uc($image) : uc("$image-$sku") ;
```

Reply:
done